src/components/ipDetector.ts

```diff
@@ -0,0 +1,122 @@
+import { EventEmitter } from 'events';
+import IDetector from './../interfaces/IDetector';
+import IIpDeviceManager from '../interfaces/iIpDeviceManager';
+import DetectorOpts from './../interfaces/IDetectorOpts';
+import CameraEvents from './../utilitis/events';
+import DetectionsConverter from './../utilitis/detectionsConverter';
+import * as huddly from '@huddly/camera-proto/lib/api/huddly_pb';
+import Logger from './../utilitis/logger';
+import { Empty } from 'google-protobuf/google/protobuf/empty_pb';
+
+const PREVIEW_IMAGE_SIZE = { width: 832, height: 480 };
+
+export default class IpDetector extends EventEmitter implements IDetector {
+  _deviceManager: IIpDeviceManager;
+  _detectionHandler: any;
+  _framingHandler: any;
+  _frame: any;
+  _options: DetectorOpts;
+  _intervalId: NodeJS.Timeout;
+  _defaultLabelWhiteList: Array<String> = ['head', 'person'];
+  _subscriptionsSetup: boolean = false;
+  _detectorInitialized: boolean = false;
+  _previewStreamStarted: boolean = false;
+  _usePeopleCount: boolean = false;
+  _UPDATE_INTERVAL: number = 500; // ms
```

Comment:
Could decrease this. Might be good with the possible improvements to the preview stream fps?